Client/Public/JS/Controllers/GameController.js

```diff
@@ -86,19 +87,24 @@ function initModels() {
     // Init the jobs model.
     modelJobs['resourceCollector'] = 
         {
-            id: document.getElementById('jobResourceCollector').id,
-            allocateButton: document.getElementById('jobResourceCollector').children[2],
-            deallocateButton: document.getElementById('jobResourceCollector').children[3]
+            id: 'jobResourceCollector',
+            resourceType: ['resource'],
+            velocity: {'resource': modelResourceRates['resourceCollector'] * modelResource['townspeopleResourceCollector']},
+            allocateButton: 'resourceCollectorAddWorker',
```

Comment:
NTS-2: Probably rename these 'allocateButtonId' and 'deallocateButtonId'. And any other place that specifies an id.